oil_lang/expr_to_ast.py

```diff
@@ -914,9 +914,75 @@ def Proc(self, pnode):
 
     return sig
 
+  def func_item(self, node):
+    """
+    func_item: (
+      ('var' | 'const') name_type_list '=' testlist  # oil_var_decl
+
+      # TODO: for, if/switch, with, break/continue/return, try/throw, etc.
+    | 'while' test suite
+    | 'for' name_type_list 'in' test suite
+    | flow_stmt
+    | 'set' place_list (augassign | '=') testlist  # oil_place_mutation
+      # x  f(x)  etc.
+      #
+      # And x = 1.  Python uses the same "hack" to fit within pgen2.  It also
+      # supports a = b = 1, which we don't want.
+      #
+      # And echo 'hi' 'there'
+      #
+      # TODO: expr_to_ast needs to validate this
+    | testlist (['=' testlist] | tea_word*)
+    )
+    """
+    if node.tok.id == Id.Expr_While:
+      return command.While(self.Expr(node.children[1]), self._Suite(node.children[2]))
+    elif node.tok.id == Id.Expr_For:
+      return command.For(
+          self._NameTypeList(node.children[1]),
+          self.Expr(node.children[3]),
+          self._Suite(node.children[4])
+        )
+    elif node.tok.id == Id.Expr_Break:
+      return command.Break()
+    elif node.tok.id == Id.Expr_Continue:
+      return command.Continue()
+    elif node.tok.id == Id.Expr_Return:
+      # 'return' [testlist]
+      if len(node.children) == 1:
+        return command.TeaReturn()
+      else:
+        return command.TeaReturn(self.Expr(node.children[1]))
+    else:
+      raise NotImplementedError(node)
+
+  def func_items(self, pnode):
+    # type: (PNode) -> List[ast.stmt]
+    """
+    func_items: func_item (semi_newline func_item)* [semi_newline]
+    """
+    raw_items = pnode.children
+    if raw_items[-1].typ == grammar_nt.semi_newline:
```

Comment:
I'm not sure this is necessary, since `::2` should do the right thing?

but yeah this code is inherently sort of fiddly that way so I think I will just merge and see what happens

Reply:
> I'm not sure this is necessary, since ::2 should do the right thing?

Well, since the pattern  `x, y, x, y, x, ...` is ensured; taking first, third, fifth item seemed make sense. Of course we could possibly right it in tons of different ways, including more explicit ones. But for this case, it seemed simple and powerful enough.